File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int low=0,high=nums.length-1;
        
        while (low<high)
        {
            int mid = 2*((low+high)/4);
            if(nums[mid]==nums[mid+1])
                low=mid+2;
            else 
                high=mid;
        }
        return nums[low];
    }
}

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        l = 0
        r = n - 1
        while(l < r):
            mid = 2*((l + r) // 4)
            if nums[mid] != nums[mid+1]:
                r = mid
            else:
                l = mid + 2
        return nums[r]
        